### ISSUE TITLE
3D selection issues

### ISSUE THREAD

**Issue by Zylann** [godotengine/godot#10979]
Godot 3.0 master
Windows 10 64 bits

I ran into a few issues with selection behaviour in the 3D editor.
The test done is to recreate this scene mesh by mesh (ProceduralMeshes):

![image](https://user-images.githubusercontent.com/1311555/30038442-e684f9a6-91c4-11e7-8095-ebf22f44d198.png)

1) I was unable to select the vertical quad in any way.

2) I was also unable to select the horizontal plane or the cube, but after some delay they became selectable, not sure why. That delay took between 1 to 30 seconds (while I was creating the scene mesh by mesh)

3) Rectangular selection doesn't select anything Fixed by #10897, but reappeared in alpha2

4) If done from lower-right to top-left, rectangular selection draws the rectangle backwards (and still doesn't select anything)

5) If I select a Spatial item, I can't select any other in the viewport even when clicking right on it. If I drag the mouse not on the axes, it still drags the object even though I'm not touching any gizmo.

6) Add a MeshInstance to the scene, assign it a mesh (a cube .obj). Notice you can't select it at all, unless you close the scene and reopen it.

**Comment by mrcdk** [comment]
1. I can confirm this still happens in a build from 308a6f4f99a5204682a2612d097e5c066f88196f
2. This seem to happen the first time you add the mesh to the MeshInstance. It won't be selectable inside the viewport until you select another MeshInstance inside the viewport. It doesn't happen if you duplicate an already created MeshInstance even if this MeshInstance wasn't selectable before. QuadMesh is still not selectable.
3. Will only select spatials with collisions_segments because of this:
- Inside this function: https://github.com/godotengine/godot/blob/master/editor/plugins/spatial_editor_plugin.cpp#L435 this 

```
if (seg->intersect_frustum(camera, frustum))
     _select(sp, true, false);
```
will call this function https://github.com/godotengine/godot/blob/master/editor/spatial_editor_gizmos.cpp#L326 which only checks for the `collision_segments` but doesn't check the `collision_mesh` (which is what those meshes use)

I don't know how to fix that. I commented out the `if (seg->intersect_frustum(camera, frustum))` and it worked but I don't know if this is a correct fix or not. Hopefully someone else can help here.

4. is https://github.com/godotengine/godot/issues/10936

**Comment by Zylann** [comment]
I build a more recent Godot version, and another issue is adding to the count:

5) If I select a Spatial item, I can't select any other in the viewport even when clicking right on it. If I drag the mouse not on the axes, it still drags the object even though I'm not touching any gizmo.

**Comment by Zylann** [comment]
Found another one:

6) Add a MeshInstance to the scene, assign it a mesh (a cube .obj). Notice you can't select it at all, unless you close the scene and reopen it.

**Comment by akien-mga** [comment]
See also https://youtu.be/lRbefcsYDi4?t=55 from #12545.

**Comment by akien-mga** [comment]
There were various fixes to select issues in the 3D viewport, which parts of this issue are still relevant? If not all of them, it might be worth opening new tickets for each problem and close this one.

**Comment by samvila** [comment]
@akien-mga 
I just tried the RC1 on Windows 10, creating a few boxes using the meshinstancer node and selecting them using the rectangle selection tool and it doesn't work so I guess the latest fixes were not enough to handle this problem.

**Comment by akien-mga** [comment]
Fixed by #19487.

**Comment by akien-mga** [comment]
Actually #19487 may have fixed only part of those issues - but it would be good to put the remaining ones each in their own issue for better visibility on their resolution.
